python/dgl/backend/pytorch/sparse.py

```diff
@@ -834,7 +834,22 @@ def csrmask(gidxA, A_weights, gidxB):
     return CSRMask.apply(gidxA, A_weights, gidxB)
 
 def segment_mm(A, B, seglen_A):
+    if str(A.device) == "cpu":
```

Comment:
add sanity checks such as `A.shape[0] == th.sum(seglen_A)` in case of access out of bound?